src/exchange/exchange.py

```diff
@@ -45,11 +45,16 @@ class Exchange:
     provider: Provider
     model: str
     system: str
-    moderator: Moderator = field(default=ContextTruncate())
+    moderator: Moderator = field(default=None)
     tools: Tuple[Tool] = field(factory=tuple, converter=tuple)
     messages: List[Message] = field(factory=list)
     checkpoint_data: CheckpointData = field(factory=CheckpointData)
 
+    def __attrs_post_init__(self):
+        """Ensures context truncation uses the same model as the exchange"""
```

Comment:
@michaelneale this part is very important, as otherwise what was happening was we were using ContextTruncate's default model which doesn't match the exchange, so messes up truncation.

Reply:
is this a current bug or is it being overwritten elsewhere?

Reply:
it is a bug but since probably most are testing where gpt-4 is available it doesn't pick up

Reply:
I think in reality truncate is most often used

Reply:
yep main thing this does is make sure that when no moderator is supplied, it is backfilled with truncate, but not using the hard-coded model in the ContextTruncate constructor.

https://github.com/square/exchange/blob/8239beb59be15462967887e6929e6caf363637d5/src/exchange/moderators/truncate.py#L23

Instead, use the same model as the user passed.

Reply:
https://github.com/square/exchange/pull/35 - I think truncate shouldn't have that default - would that simplify it?

Reply:
yep. removed this topic for now!